File: rb/22.rb
def name_value name
  name.chars.map { |c| c.ord - 64 }.reduce(:+)
end

p IO.read('../input/22')
  .scan(/\w+/)
  .sort
  .each_with_index
  .reduce(0) { |sum, (name, i)|
    sum += name_value(name) * i.succ
  }

File: py/22.py
from re import findall

def value(name):
    return reduce(lambda a, b: a + b,
           map(lambda c: ord(c) - 64, list(name)))

with open('../input/22') as f:
    names = sorted(findall('\w+', f.read()))
    print sum(value(names[i]) * (i + 1) for i in xrange(len(names)))
